packages/alpinejs/src/mutation.js

```diff
@@ -194,14 +194,15 @@ function onMutate(mutations) {
         node._x_ignore = true
     })
     for (let node of addedNodes) {
-        // If an element gets moved on a page, it's registered
-        // as both an "add" and "remove", so we want to skip those.
-        if (removedNodes.includes(node)) continue
+        // If the node is already init, it means it's a move operation
+        if (node._x_isInit) continue
 
         // If the node was eventually removed as part of one of his
         // parent mutations, skip it
         if (! node.isConnected) continue
 
+        node._x_isInit = true
```

Comment:
This is the incorrect place to handle this, as this does not actually do any initializing.

Reply:
Do you mean something like 

```js
export function initTree(el, walker = walk, intercept = () => {}) {
    deferHandlingDirectives(() => {
        walker(el, (el, skip) => {
            intercept(el, skip)

            initInterceptors.forEach(i => i(el, skip))

            directives(el, el.attributes).forEach(handle => handle())

            if (el._x_ignore) {
                skip()
            } else {
                el._x_isInit = true;
            }
        })
    })
}
```

I'm really newbie for the Alpine code source ;)

Reply:
Almost. More like on every walk of an element. This should only do it on tree roots.

Reply:
hum, you mean in the `walker` callback ?
```diff
export function initTree(el, walker = walk, intercept = () => {}) {
    deferHandlingDirectives(() => {
        walker(el, (el, skip) => {
+           el._x_isInit = true
+
            intercept(el, skip)

            initInterceptors.forEach(i => i(el, skip))

            directives(el, el.attributes).forEach(handle => handle())
            el._x_ignore && skip()
        })
    })
}
```

the previous code works well for the moment, I could try this one too if you want

Reply:
Okay I get it, I simply do this

```js
export function initTree(el, walker = walk, intercept = () => {}) {
    deferHandlingDirectives(() => {
        walker(el, (el, skip) => {
            if (el._x_isInit) return;
            el._x_isInit = true
            // ...
```

in the lifecycle and everything works great, we have a couple of edge case they all works perfectly \o/

Reply:
hum that's strange cause there is a test that failed, however everything works. There is something that I don't catch